FileIO/Legacy/MeshIO.cpp

```diff
@@ -133,18 +148,22 @@ MeshLib::Mesh* MeshIO::loadMeshFromFile(const std::string& file_name)
 	}
 }
 
-MeshLib::Element* MeshIO::readElement(const std::string& line,
-                                      const std::vector<MeshLib::Node*> &nodes)
+std::size_t MeshIO::readMaterialID(std::istream & in) const
+{
+	unsigned index, patch_index;
+	in >> index >> patch_index;
+	return patch_index;
+}
+
+MeshLib::Element* MeshIO::readElement(std::istream& in,
+	const std::vector<MeshLib::Node*> &nodes) const
 {
-	std::stringstream ss (line);
 	std::string elem_type_str("");
 	MeshElemType elem_type (MeshElemType::INVALID);
-	unsigned index, patch_index;
-	ss >> index >> patch_index;
 
 	do {
-		ss >> elem_type_str;
-		if (ss.fail())
+		in >> elem_type_str;
+		if (in.fail())
 			return NULL;
 		elem_type = String2MeshElemType(elem_type_str);
 	} while (elem_type == MeshElemType::INVALID);
```

Comment:
sorry for the many questions but why do we need a ```do ... while``` here? (i see it is old code and presumably there is a reason for this but i cannot think of any ...)